osf/metrics/reporters/spam_count.py

```diff
@@ -0,0 +1,84 @@
+        target_month = report_yearmonth.target_month()
+        next_month = report_yearmonth.next_month()
+
+        report = SpamSummaryReport(
+            report_yearmonth=str(report_yearmonth),
+            # Node Log entries
+            confirmed_spam_node=NodeLog.objects.filter(
+                action=NodeLog.CONFIRM_SPAM,
+                created__gt=target_month,
+                created__lt=next_month,
+                node__type='osf.node',
+            ).count(),
+            nodes_confirmed_ham=NodeLog.objects.filter(
+                action=NodeLog.CONFIRM_HAM,
+                created__gt=target_month,
+                created__lt=next_month,
+                node__type='osf.node',
+            ).count(),
+            nodes_flagged=NodeLog.objects.filter(
+                action=NodeLog.FLAG_SPAM,
+                created__gt=target_month,
+                created__lt=next_month,
+                node__type='osf.node',
+            ).count(),
+            # Registration Log entries
+            registration_confirmed_spam=NodeLog.objects.filter(
+                action=NodeLog.CONFIRM_SPAM,
+                created__gt=target_month,
+                created__lt=next_month,
+                node__type='osf.registration',
+            ).count(),
+            registration_confirmed_ham=NodeLog.objects.filter(
+                action=NodeLog.CONFIRM_HAM,
+                created__gt=target_month,
+                created__lt=next_month,
+                node__type='osf.registration',
+            ).count(),
+            registration_flagged=NodeLog.objects.filter(
+                action=NodeLog.FLAG_SPAM,
+                created__gt=target_month,
+                created__lt=next_month,
+                node__type='osf.registration',
+            ).count(),
+            # Preprint Log entries
+            preprint_confirmed_spam=PreprintLog.objects.filter(
+                action=PreprintLog.CONFIRM_SPAM,
+                created__gt=target_month,
+                created__lt=next_month,
+            ).count(),
+            preprint_confirmed_ham=PreprintLog.objects.filter(
+                action=PreprintLog.CONFIRM_HAM,
+                created__gt=target_month,
+                created__lt=next_month,
+            ).count(),
+            preprint_flagged=PreprintLog.objects.filter(
+                action=PreprintLog.FLAG_SPAM,
+                created__gt=target_month,
+                created__lt=next_month,
+            ).count(),
+            # New Users marked as Spam/Ham
+            users_marked_as_spam=OSFUser.objects.filter(
+                spam_status=SpamStatus.SPAM,
+                created__gt=target_month,
+                created__lt=next_month,
+            ).count(),
+            user_marked_as_ham=OSFUser.objects.filter(
+                spam_status=SpamStatus.HAM,
+                created__gt=target_month,
+                created__lt=next_month,
+            ).count()
```

Comment:
How can we clarify that this is the "number of users created in the target month that were marked as Ham/Spam" and not "The number of users marked as Ham/Spam in the last month"? That's a behavior difference with the other resource types that needs to be clear

Reply:
In the long view, let's consider adding a `date_reenabled` or similar to the User Model. Then we can use `date_disabled` and `date_reenabled` along with the `spam_status` as an invariant for users confirmed one way or the other in the given month.